main/observing/condition_checker.py

```diff
@@ -212,7 +212,9 @@ def rain_check(self):
                 urllib3.exceptions.InvalidHeader, requests.exceptions.ConnectionError, requests.exceptions.Timeout,
                 requests.exceptions.HTTPError):
             return None
-        api_key = re.search(r'"SUN_V3_API_KEY":"(.+?)",', self.radar.text).group(1)
+
+        #api_key = re.search(r'"SUN_V3_API_KEY":"(.+?)",', self.radar.text).group(1)
+        api_key = self.radar.text.split(r'"SUN_V3_API_KEY_CLIENT\":\"')[1].split(r'\",\"WX_API_PROTOCOL_CLIENT\"')[0]
```

Comment:
This might work for now, but it's prone to the same kind of failure in the future if the html ever changes again.  The way I mentioned isn't perfect either, but at least it has a bit more flexibility in case the text changes again in SUN_V3_API_KEY_CLIENT